engine/include/Engine/Intermediate/GameObject.hpp

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include "Engine/Intermediate/TransformComponent.hpp" //TransformComponent
+#include <list>                                       //std::list
+#include <memory>                                     //std::unique_ptr
+#include <string>                                     //std::string
+#include <vector>                                     //std::vector
+
+// in Inl
+#include "Engine/Intermediate/DataChunk.hpp"
+
+namespace GPE
+{
+class Component;
+
+class GameObject
+{
+public:
+    struct CreateArg
+    {
+        std::string                   name = "";
+        TransformComponent::CreateArg transformArg;
+    };
+
+protected:
+    std::string         m_name;
+    TransformComponent* m_pTransform;
+
+    std::list<Component*> m_pComponents;
+    std::string           m_tag{"GameObject"};
+    bool m_isDead{false}; // Flag that inform it parent that this transform must be destroy on update loop
+
+public:
+    GameObject*                            parent   = nullptr;
+    std::list<std::unique_ptr<GameObject>> children = {};
+
+public: // TODO : Protected method ?
+    template <typename T>
+    void updateComponentLink(const T* oldPtr, T* newPtr) noexcept;
+
+public:
+    inline
+    GameObject(const CreateArg& arg);
```

Comment:
**Suggestion**

I still believe creating an intermediate `struct` in this context is not necessary, and just makes code harder to read and understand. I would *much* rather write/read:
```C++
GameObject myObj{"character_1", {50.f, 0.f, 25.f}, Vec3::zero(), {2.f}};
```
or even
```C++
GameObject myObj.setName()
                .setPosition({50.f, 0.f, 25.f})
                .setScale({2.f});
```
than
```C++
GameObject myObj{{"character_1", {{50.f, 0.f, 25.f}, Vec3::zero(); {2.f}}}};
```

If we go beyond 5-6 parameters, I would totally understand the use of a `struct`.
As long as it is not the case, I doubt we will get any value out of using these intermediate `struct`s, quite the opposite

Reply:
We don't know if gameObject will need more parameter in the future, keep struct to encapsule arg allow us to upgrade code more esaly. It also allow to give a landmark for the user (Componenent and entity is create thanks to struct like vulkane)